cpp/src/arrow/flight/sql/odbc/flight_sql/accessors/timestamp_array_accessor_test.cc

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/flight/sql/odbc/flight_sql/accessors/timestamp_array_accessor.h"
+#include "arrow/flight/sql/odbc/flight_sql/utils.h"
+#include "arrow/testing/builder.h"
+#include "gtest/gtest.h"
+#include "odbcabstraction/calendar_utils.h"
+
+namespace driver {
+namespace flight_sql {
+
+using arrow::ArrayFromVector;
+using arrow::TimestampType;
+using arrow::TimeUnit;
+
+using odbcabstraction::OdbcVersion;
+using odbcabstraction::TIMESTAMP_STRUCT;
+
+using odbcabstraction::GetTimeForSecondsSinceEpoch;
+
+TEST(TEST_TIMESTAMP, TIMESTAMP_WITH_MILLI) {
```

Comment:
FWIW, the test suite/case name are effectively class names, so you'd normally expect to see `TestTimestamp, TimestampWithMilli`